### ISSUE TITLE
Protocol-related performance regression

### ISSUE THREAD

**Issue by arnimarj** [agronholm/typeguard#104]
Hi,

We recently rolled out the 2.7.0 release of typeguard across our infrastructure, and noticed an interesting performance regression w.r.t. 2.6.1. We've narrowed it down to the new protocol support, and in particular the call to `inspect.getattr_static`.

A small example to demonstrate:

```bash
$ pip install -q typeguard==2.6.1
$ python guard.py
T: 3.40 seconds
version: 2.6.1
$ 
$ pip install -q typeguard==2.7.0
$ python guard.py 
T: 11.15 seconds
version: 2.7.0
```

`guard.py`:

```python
import attr
import time
import typeguard


@attr.s
class MyTypedClass:
	a: str = attr.ib()
	b: str = attr.ib()
	c: int = attr.ib()

	def __attrs_post_init__(self) -> None:
		TypecheckAttrInstance(self)


def TypecheckAttrInstance(obj):
	assert(attr.has(obj))

	for a in attr.fields(type(obj)):
		value = getattr(obj, a.name)
		typeguard.check_type(a.name, value, a.type)


def main():
	t_0 = time.monotonic()

	for _ in range(10**5 * 4):
		MyTypedClass(a='foo', b='bar', c=42)

	print(f'T: {time.monotonic() - t_0:.2f} seconds')
	print(f'version: {_typeguard_version()}')


def _typeguard_version():
	from pip._internal.utils.misc import get_installed_distributions

	for p in get_installed_distributions():
		if 'typeguard' in str(p):
			return p.version

	raise ValueError


if __name__ == '__main__':
	main()
```

**Comment by agronholm** [comment]
Hard to believe that `getattr_static()` would cause the performance to drop to 1/3 of the original, but I will try to reproduce these results.

**Comment by agronholm** [comment]
I can repro the results:
```
T: 18.00 seconds
version: 2.7.0
T: 8.06 seconds
version: 2.6.1
```

**Comment by agronholm** [comment]
After commenting out the Protocol check, it's back to 8 seconds.

**Comment by agronholm** [comment]
Replacing it with `getattr()` restored most of the lost performance. It's still some 200-400ms slower than in 2.6.1, but it's more reasonable now. I'm going to do this replacement in the actual code now.

**Comment by agronholm** [comment]
I've now released v2.7.1 which includes this fix.
